### DISCUSSION TITLE
API Problems

### DISCUSSION THREAD

**Discussion by zenotornado** [wavelog/wavelog#596] [General]
Hello folks,

I've a problem getting my API runnig. I made a PHP Form for fast logging from my mobile. But I don't get it into wavelog. If I use an Online API Test I get the landing page as a result. No failure, no message... (the key is changed for the discussion ;-) )

<img width="602" alt="API result" src="https://github.com/user-attachments/assets/f884a74f-37d9-4053-9740-67fa90e99565">

Any ideas to get this running?

**Comment by phl0** [comment]
What does the logfile of your Wavelog istance say? Please also check the webservers log if possible. You may need to adjust the log threshold in application/config/config.php.

**Comment by zenotornado** [comment]
Logs are set on "1" in Config... I only get this result:

<html>
<head>
	<title>403 Forbidden</title>
</head>
<body>

<p>Directory access is forbidden.</p>

</body>
</html>

No other log filex

**Reply by phl0** [reply]
Did you access the log file directory via browser?

**Comment by zenotornado** [comment]
Checked is with FTP. There are only the file generatd while installing.

**Reply by zenotornado** [reply]
Have seen your name in your profile... 73, DO1JOG :-)

**Reply by phl0** [reply]
Set the log threshold to 4 and look for the logfile dated today in application/logs

**Reply by zenotornado** [reply]
<?php defined('BASEPATH') OR exit('No direct script access allowed'); ?>

INFO - 2024-07-18 11:26:27 --> Config Class Initialized
INFO - 2024-07-18 11:26:27 --> Hooks Class Initialized
INFO - 2024-07-18 11:26:27 --> UTF-8 Support Enabled
INFO - 2024-07-18 11:26:27 --> Utf8 Class Initialized
INFO - 2024-07-18 11:26:27 --> URI Class Initialized
DEBUG - 2024-07-18 11:26:27 --> No URI present. Default controller set.
INFO - 2024-07-18 11:26:27 --> Router Class Initialized
INFO - 2024-07-18 11:26:27 --> Output Class Initialized
INFO - 2024-07-18 11:26:27 --> Security Class Initialized
INFO - 2024-07-18 11:26:27 --> Input Class Initialized
INFO - 2024-07-18 11:26:27 --> Language Class Initialized
INFO - 2024-07-18 11:26:27 --> Config file loaded: /homepages/1/d1014508302/htdocs/logbook/wavelog-master/application/config/gettext.php
INFO - 2024-07-18 11:26:27 --> Loader Class Initialized
INFO - 2024-07-18 11:26:27 --> Config file loaded: /homepages/1/d1014508302/htdocs/logbook/wavelog-master/application/config/wavelog.php
INFO - 2024-07-18 11:26:27 --> Config file loaded: /homepages/1/d1014508302/htdocs/logbook/wavelog-master/application/config/bands.php
INFO - 2024-07-18 11:26:27 --> Config file loaded: /homepages/1/d1014508302/htdocs/logbook/wavelog-master/application/config/lotw.php
INFO - 2024-07-18 11:26:27 --> Helper loaded: url_helper
INFO - 2024-07-18 11:26:27 --> Helper loaded: security_helper
INFO - 2024-07-18 11:26:27 --> Helper loaded: language_helper
INFO - 2024-07-18 11:26:27 --> Database Driver Class Initialized
INFO - 2024-07-18 11:26:27 --> Session: Class initialized using 'files' driver.
INFO - 2024-07-18 11:26:27 --> Migrations Class Initialized
INFO - 2024-07-18 11:26:27 --> Language file loaded: language/english/migration_lang.php
INFO - 2024-07-18 11:26:27 --> Database Forge Class Initialized
INFO - 2024-07-18 11:26:27 --> Model "Options_model" initialized
INFO - 2024-07-18 11:26:27 --> Model "User_options_model" initialized
INFO - 2024-07-18 11:26:27 --> Model "Logbooks_model" initialized
INFO - 2024-07-18 11:26:27 --> Model "Stations" initialized
INFO - 2024-07-18 11:26:27 --> Model "Oqrs_model" initialized
INFO - 2024-07-18 11:26:27 --> Controller Class Initialized
INFO - 2024-07-18 11:26:27 --> Model "User_Model" initialized
INFO - 2024-07-18 11:26:27 --> Config Class Initialized
INFO - 2024-07-18 11:26:27 --> Hooks Class Initialized
INFO - 2024-07-18 11:26:27 --> UTF-8 Support Enabled
INFO - 2024-07-18 11:26:27 --> Utf8 Class Initialized
INFO - 2024-07-18 11:26:27 --> URI Class Initialized
INFO - 2024-07-18 11:26:27 --> Router Class Initialized
INFO - 2024-07-18 11:26:27 --> Output Class Initialized
INFO - 2024-07-18 11:26:27 --> Security Class Initialized
INFO - 2024-07-18 11:26:27 --> Input Class Initialized
INFO - 2024-07-18 11:26:27 --> Language Class Initialized
INFO - 2024-07-18 11:26:27 --> Config file loaded: /homepages/1/d1014508302/htdocs/logbook/wavelog-master/application/config/gettext.php
INFO - 2024-07-18 11:26:27 --> Loader Class Initialized
INFO - 2024-07-18 11:26:27 --> Config file loaded: /homepages/1/d1014508302/htdocs/logbook/wavelog-master/application/config/wavelog.php
INFO - 2024-07-18 11:26:27 --> Config file loaded: /homepages/1/d1014508302/htdocs/logbook/wavelog-master/application/config/bands.php
INFO - 2024-07-18 11:26:27 --> Config file loaded: /homepages/1/d1014508302/htdocs/logbook/wavelog-master/application/config/lotw.php
INFO - 2024-07-18 11:26:27 --> Helper loaded: url_helper
INFO - 2024-07-18 11:26:27 --> Helper loaded: security_helper
INFO - 2024-07-18 11:26:27 --> Helper loaded: language_helper
INFO - 2024-07-18 11:26:27 --> Database Driver Class Initialized
INFO - 2024-07-18 11:26:27 --> Session: Class initialized using 'files' driver.
INFO - 2024-07-18 11:26:27 --> Migrations Class Initialized
INFO - 2024-07-18 11:26:27 --> Language file loaded: language/english/migration_lang.php
INFO - 2024-07-18 11:26:27 --> Database Forge Class Initialized
INFO - 2024-07-18 11:26:27 --> Model "Options_model" initialized
INFO - 2024-07-18 11:26:27 --> Model "User_options_model" initialized
INFO - 2024-07-18 11:26:27 --> Model "Logbooks_model" initialized
INFO - 2024-07-18 11:26:27 --> Model "Stations" initialized
INFO - 2024-07-18 11:26:27 --> Model "Oqrs_model" initialized
INFO - 2024-07-18 11:26:27 --> Controller Class Initialized
DEBUG - 2024-07-18 11:26:27 --> Migration class already loaded. Second attempt ignored.
INFO - 2024-07-18 11:26:27 --> Model "User_Model" initialized
INFO - 2024-07-18 11:26:27 --> Helper loaded: form_helper
INFO - 2024-07-18 11:26:27 --> Form Validation Class Initialized
INFO - 2024-07-18 11:26:27 --> File loaded: /homepages/1/d1014508302/htdocs/logbook/wavelog-master/application/views/interface_assets/mini_header.php
INFO - 2024-07-18 11:26:27 --> Model "Themes_model" initialized
INFO - 2024-07-18 11:26:27 --> File loaded: /homepages/1/d1014508302/htdocs/logbook/wavelog-master/application/views/layout/messages.php
INFO - 2024-07-18 11:26:27 --> File loaded: /homepages/1/d1014508302/htdocs/logbook/wavelog-master/application/views/user/login.php
INFO - 2024-07-18 11:26:27 --> Model "Qsl_model" initialized
INFO - 2024-07-18 11:26:27 --> Can not get qsl card image path because no user_id in session data
INFO - 2024-07-18 11:26:27 --> Model "Eqsl_images" initialized
INFO - 2024-07-18 11:26:27 --> Can not get eqsl image path because no user_id in session data
INFO - 2024-07-18 11:26:27 --> Can not get qsl card image path because no user_id in session data
INFO - 2024-07-18 11:26:27 --> File loaded: /homepages/1/d1014508302/htdocs/logbook/wavelog-master/application/views/interface_assets/footer.php
INFO - 2024-07-18 11:26:27 --> Final output sent to browser
INFO - 2024-07-18 11:26:27 --> Total execution time: 0.0391
INFO - 2024-07-18 11:26:28 --> Config Class Initialized
INFO - 2024-07-18 11:26:28 --> Hooks Class Initialized
INFO - 2024-07-18 11:26:28 --> UTF-8 Support Enabled
INFO - 2024-07-18 11:26:28 --> Utf8 Class Initialized
INFO - 2024-07-18 11:26:28 --> URI Class Initialized
DEBUG - 2024-07-18 11:26:28 --> No URI present. Default controller set.
INFO - 2024-07-18 11:26:28 --> Router Class Initialized
INFO - 2024-07-18 11:26:28 --> Output Class Initialized
INFO - 2024-07-18 11:26:28 --> Security Class Initialized
INFO - 2024-07-18 11:26:28 --> Input Class Initialized
INFO - 2024-07-18 11:26:28 --> Language Class Initialized
INFO - 2024-07-18 11:26:28 --> Config file loaded: /homepages/1/d1014508302/htdocs/logbook/wavelog-master/application/config/gettext.php
INFO - 2024-07-18 11:26:28 --> Loader Class Initialized
INFO - 2024-07-18 11:26:28 --> Config file loaded: /homepages/1/d1014508302/htdocs/logbook/wavelog-master/application/config/wavelog.php
INFO - 2024-07-18 11:26:28 --> Config file loaded: /homepages/1/d1014508302/htdocs/logbook/wavelog-master/application/config/bands.php
INFO - 2024-07-18 11:26:28 --> Config file loaded: /homepages/1/d1014508302/htdocs/logbook/wavelog-master/application/config/lotw.php
INFO - 2024-07-18 11:26:28 --> Helper loaded: url_helper
INFO - 2024-07-18 11:26:28 --> Helper loaded: security_helper
INFO - 2024-07-18 11:26:28 --> Helper loaded: language_helper
INFO - 2024-07-18 11:26:28 --> Database Driver Class Initialized
INFO - 2024-07-18 11:26:28 --> Session: Class initialized using 'files' driver.
INFO - 2024-07-18 11:26:28 --> Migrations Class Initialized
INFO - 2024-07-18 11:26:28 --> Language file loaded: language/english/migration_lang.php
INFO - 2024-07-18 11:26:28 --> Database Forge Class Initialized
INFO - 2024-07-18 11:26:28 --> Model "Options_model" initialized
INFO - 2024-07-18 11:26:28 --> Model "User_options_model" initialized
INFO - 2024-07-18 11:26:28 --> Model "Logbooks_model" initialized
INFO - 2024-07-18 11:26:28 --> Model "Stations" initialized
INFO - 2024-07-18 11:26:28 --> Model "Oqrs_model" initialized
INFO - 2024-07-18 11:26:28 --> Controller Class Initialized
INFO - 2024-07-18 11:26:28 --> Model "User_Model" initialized
INFO - 2024-07-18 11:26:28 --> Config Class Initialized
INFO - 2024-07-18 11:26:28 --> Hooks Class Initialized
INFO - 2024-07-18 11:26:28 --> UTF-8 Support Enabled
INFO - 2024-07-18 11:26:28 --> Utf8 Class Initialized
INFO - 2024-07-18 11:26:28 --> URI Class Initialized
INFO - 2024-07-18 11:26:28 --> Router Class Initialized
INFO - 2024-07-18 11:26:28 --> Output Class Initialized
INFO - 2024-07-18 11:26:28 --> Security Class Initialized
INFO - 2024-07-18 11:26:28 --> Input Class Initialized
INFO - 2024-07-18 11:26:28 --> Language Class Initialized
INFO - 2024-07-18 11:26:28 --> Config file loaded: /homepages/1/d1014508302/htdocs/logbook/wavelog-master/application/config/gettext.php
INFO - 2024-07-18 11:26:28 --> Loader Class Initialized
INFO - 2024-07-18 11:26:28 --> Config file loaded: /homepages/1/d1014508302/htdocs/logbook/wavelog-master/application/config/wavelog.php
INFO - 2024-07-18 11:26:28 --> Config file loaded: /homepages/1/d1014508302/htdocs/logbook/wavelog-master/application/config/bands.php
INFO - 2024-07-18 11:26:28 --> Config file loaded: /homepages/1/d1014508302/htdocs/logbook/wavelog-master/application/config/lotw.php
INFO - 2024-07-18 11:26:28 --> Helper loaded: url_helper
INFO - 2024-07-18 11:26:28 --> Helper loaded: security_helper
INFO - 2024-07-18 11:26:28 --> Helper loaded: language_helper
INFO - 2024-07-18 11:26:28 --> Database Driver Class Initialized
INFO - 2024-07-18 11:26:28 --> Session: Class initialized using 'files' driver.
INFO - 2024-07-18 11:26:28 --> Migrations Class Initialized
INFO - 2024-07-18 11:26:28 --> Language file loaded: language/english/migration_lang.php
INFO - 2024-07-18 11:26:28 --> Database Forge Class Initialized
INFO - 2024-07-18 11:26:28 --> Model "Options_model" initialized
INFO - 2024-07-18 11:26:28 --> Model "User_options_model" initialized
INFO - 2024-07-18 11:26:28 --> Model "Logbooks_model" initialized
INFO - 2024-07-18 11:26:28 --> Model "Stations" initialized
INFO - 2024-07-18 11:26:28 --> Model "Oqrs_model" initialized
INFO - 2024-07-18 11:26:28 --> Controller Class Initialized
DEBUG - 2024-07-18 11:26:28 --> Migration class already loaded. Second attempt ignored.
INFO - 2024-07-18 11:26:28 --> Model "User_Model" initialized
INFO - 2024-07-18 11:26:28 --> Helper loaded: form_helper
INFO - 2024-07-18 11:26:28 --> Form Validation Class Initialized
INFO - 2024-07-18 11:26:28 --> File loaded: /homepages/1/d1014508302/htdocs/logbook/wavelog-master/application/views/interface_assets/mini_header.php
INFO - 2024-07-18 11:26:28 --> Model "Themes_model" initialized
INFO - 2024-07-18 11:26:28 --> File loaded: /homepages/1/d1014508302/htdocs/logbook/wavelog-master/application/views/layout/messages.php
INFO - 2024-07-18 11:26:28 --> File loaded: /homepages/1/d1014508302/htdocs/logbook/wavelog-master/application/views/user/login.php
INFO - 2024-07-18 11:26:28 --> Model "Qsl_model" initialized
INFO - 2024-07-18 11:26:28 --> Can not get qsl card image path because no user_id in session data
INFO - 2024-07-18 11:26:28 --> Model "Eqsl_images" initialized
INFO - 2024-07-18 11:26:28 --> Can not get eqsl image path because no user_id in session data
INFO - 2024-07-18 11:26:28 --> Can not get qsl card image path because no user_id in session data
INFO - 2024-07-18 11:26:28 --> File loaded: /homepages/1/d1014508302/htdocs/logbook/wavelog-master/application/views/interface_assets/footer.php
INFO - 2024-07-18 11:26:28 --> Final output sent to browser
INFO - 2024-07-18 11:26:28 --> Total execution time: 0.0265
INFO - 2024-07-18 11:26:29 --> Config Class Initialized
INFO - 2024-07-18 11:26:29 --> Hooks Class Initialized
INFO - 2024-07-18 11:26:29 --> UTF-8 Support Enabled
INFO - 2024-07-18 11:26:29 --> Utf8 Class Initialized
INFO - 2024-07-18 11:26:29 --> URI Class Initialized
DEBUG - 2024-07-18 11:26:29 --> No URI present. Default controller set.
INFO - 2024-07-18 11:26:29 --> Router Class Initialized
INFO - 2024-07-18 11:26:29 --> Output Class Initialized
INFO - 2024-07-18 11:26:29 --> Security Class Initialized
INFO - 2024-07-18 11:26:29 --> Input Class Initialized
INFO - 2024-07-18 11:26:29 --> Language Class Initialized
INFO - 2024-07-18 11:26:29 --> Config file loaded: /homepages/1/d1014508302/htdocs/logbook/wavelog-master/application/config/gettext.php
INFO - 2024-07-18 11:26:29 --> Loader Class Initialized
INFO - 2024-07-18 11:26:29 --> Config file loaded: /homepages/1/d1014508302/htdocs/logbook/wavelog-master/application/config/wavelog.php
INFO - 2024-07-18 11:26:29 --> Config file loaded: /homepages/1/d1014508302/htdocs/logbook/wavelog-master/application/config/bands.php
INFO - 2024-07-18 11:26:29 --> Config file loaded: /homepages/1/d1014508302/htdocs/logbook/wavelog-master/application/config/lotw.php
INFO - 2024-07-18 11:26:29 --> Helper loaded: url_helper
INFO - 2024-07-18 11:26:29 --> Helper loaded: security_helper
INFO - 2024-07-18 11:26:29 --> Helper loaded: language_helper
INFO - 2024-07-18 11:26:29 --> Database Driver Class Initialized
INFO - 2024-07-18 11:26:29 --> Session: Class initialized using 'files' driver.
INFO - 2024-07-18 11:26:29 --> Migrations Class Initialized
INFO - 2024-07-18 11:26:29 --> Language file loaded: language/english/migration_lang.php
INFO - 2024-07-18 11:26:29 --> Database Forge Class Initialized
INFO - 2024-07-18 11:26:29 --> Model "Options_model" initialized
INFO - 2024-07-18 11:26:29 --> Model "User_options_model" initialized
INFO - 2024-07-18 11:26:29 --> Model "Logbooks_model" initialized
INFO - 2024-07-18 11:26:29 --> Model "Stations" initialized
INFO - 2024-07-18 11:26:29 --> Model "Oqrs_model" initialized
INFO - 2024-07-18 11:26:29 --> Controller Class Initialized
INFO - 2024-07-18 11:26:29 --> Model "User_Model" initialized
INFO - 2024-07-18 11:26:29 --> Config Class Initialized
INFO - 2024-07-18 11:26:29 --> Hooks Class Initialized
INFO - 2024-07-18 11:26:29 --> UTF-8 Support Enabled
INFO - 2024-07-18 11:26:29 --> Utf8 Class Initialized
INFO - 2024-07-18 11:26:29 --> URI Class Initialized
INFO - 2024-07-18 11:26:29 --> Router Class Initialized
INFO - 2024-07-18 11:26:29 --> Output Class Initialized
INFO - 2024-07-18 11:26:29 --> Security Class Initialized
INFO - 2024-07-18 11:26:29 --> Input Class Initialized
INFO - 2024-07-18 11:26:29 --> Language Class Initialized
INFO - 2024-07-18 11:26:29 --> Config file loaded: /homepages/1/d1014508302/htdocs/logbook/wavelog-master/application/config/gettext.php
INFO - 2024-07-18 11:26:29 --> Loader Class Initialized
INFO - 2024-07-18 11:26:29 --> Config file loaded: /homepages/1/d1014508302/htdocs/logbook/wavelog-master/application/config/wavelog.php
INFO - 2024-07-18 11:26:29 --> Config file loaded: /homepages/1/d1014508302/htdocs/logbook/wavelog-master/application/config/bands.php
INFO - 2024-07-18 11:26:29 --> Config file loaded: /homepages/1/d1014508302/htdocs/logbook/wavelog-master/application/config/lotw.php
INFO - 2024-07-18 11:26:29 --> Helper loaded: url_helper
INFO - 2024-07-18 11:26:29 --> Helper loaded: security_helper
INFO - 2024-07-18 11:26:29 --> Helper loaded: language_helper
INFO - 2024-07-18 11:26:29 --> Database Driver Class Initialized
INFO - 2024-07-18 11:26:29 --> Session: Class initialized using 'files' driver.
INFO - 2024-07-18 11:26:29 --> Migrations Class Initialized
INFO - 2024-07-18 11:26:29 --> Language file loaded: language/english/migration_lang.php
INFO - 2024-07-18 11:26:29 --> Database Forge Class Initialized
INFO - 2024-07-18 11:26:29 --> Model "Options_model" initialized
INFO - 2024-07-18 11:26:29 --> Model "User_options_model" initialized
INFO - 2024-07-18 11:26:29 --> Model "Logbooks_model" initialized
INFO - 2024-07-18 11:26:29 --> Model "Stations" initialized
INFO - 2024-07-18 11:26:29 --> Model "Oqrs_model" initialized
INFO - 2024-07-18 11:26:29 --> Controller Class Initialized
DEBUG - 2024-07-18 11:26:29 --> Migration class already loaded. Second attempt ignored.
INFO - 2024-07-18 11:26:29 --> Model "User_Model" initialized
INFO - 2024-07-18 11:26:29 --> Helper loaded: form_helper
INFO - 2024-07-18 11:26:29 --> Form Validation Class Initialized
INFO - 2024-07-18 11:26:29 --> File loaded: /homepages/1/d1014508302/htdocs/logbook/wavelog-master/application/views/interface_assets/mini_header.php
INFO - 2024-07-18 11:26:29 --> Model "Themes_model" initialized
INFO - 2024-07-18 11:26:29 --> File loaded: /homepages/1/d1014508302/htdocs/logbook/wavelog-master/application/views/layout/messages.php
INFO - 2024-07-18 11:26:29 --> File loaded: /homepages/1/d1014508302/htdocs/logbook/wavelog-master/application/views/user/login.php
INFO - 2024-07-18 11:26:29 --> Model "Qsl_model" initialized
INFO - 2024-07-18 11:26:29 --> Can not get qsl card image path because no user_id in session data
INFO - 2024-07-18 11:26:29 --> Model "Eqsl_images" initialized
INFO - 2024-07-18 11:26:29 --> Can not get eqsl image path because no user_id in session data
INFO - 2024-07-18 11:26:29 --> Can not get qsl card image path because no user_id in session data
INFO - 2024-07-18 11:26:29 --> File loaded: /homepages/1/d1014508302/htdocs/logbook/wavelog-master/application/views/interface_assets/footer.php
INFO - 2024-07-18 11:26:29 --> Final output sent to browser
INFO - 2024-07-18 11:26:29 --> Total execution time: 0.0273


Nothing about API in there...

**Reply by zenotornado** [reply]
But it will start a prot. with every API. It seems that the API request only opens the page

**Reply by phl0** [reply]
Thats weird. How exactly do you test the API? Webserver publicly available? Maybe HTTPS/SSL cert issue?

**Reply by zenotornado** [reply]
I've tried it with my own App and also with https://reqbin.com/ ... at regbin I use the demo string to be sure that it's not a problem wirh my config

{
    "key":"ABCDEFG",
    "station_profile_id":"1",
    "type":"adif",
    "string":"<call:9>N9EATaaaasent<eor>"
}

SSL is activated

**Reply by phl0** [reply]
However with log threshold there should be some output in the logfile when accessing the API ...

**Reply by phl0** [reply]
What is the exact URL you are using to access the API?

**Reply by zenotornado** [reply]
log.do1jog.de as given in Wavelog... but I'll try the direct path... hmm no... maybe I should ask the provider if there is any restriction

**Reply by phl0** [reply]
You have to add the complete path.... i.e. /index.php/api/qso for logging

**Reply by zenotornado** [reply]
Well this seems to work... haven't found it in the documentation on Git: Thank you very much

**Reply by phl0** [reply]
Thats because there are various endpoints for different purposes. See https://gitlab.com/gridtracker.org/gridtracker/-/merge_requests/316/diffs for example.

**Comment by phl0** [comment]
Can we close here?
